dpbench/infrastructure/numba_mlir_framework.py

```diff
@@ -0,0 +1,84 @@
+# Copyright 2022 Intel Corp.
+#
+# SPDX-License-Identifier: Apache 2.0
+
+import logging
+from typing import Any, Callable, Dict
+
+import numpy as np
+import pkg_resources
+
+from .framework import Framework
+
+
+class NumbaMlirFramework(Framework):
+    """A class for reading and processing framework information."""
+
+    def __init__(self, fname: str, fconfig_path: str = None):
+        """Reads framework information.
+        :param fname: The framework name.
+        """
+
+        super().__init__(fname, fconfig_path)
+
+        self.sycl_device = self.info["sycl_device"]
+
+    def imports(self) -> Dict[str, Any]:
+        """Returns a dictionary any modules and methods needed for running
+        a benchmark."""
+        import dpctl
+
+        return {"dpctl": dpctl}
```

Comment:
Do we need to import dpctl anymore? We needed it before in dpex when using dpctl.device_context().